### PR TITLE
Bugfix and other small changes to SRA SigningStage

### PR DESCRIPTION
**Pull request by gosar**
## Motivation and Context

Fix bug introduced in https://github.com/aws/aws-sdk-java-v2/pull/4144 and other small changes to SRA SigningStage

## Modifications

* Fixed bug missing `!` in shouldSign
* Wrap with updateInterceptorContext
* Added metric for signing duration

## Testing

Double checked that AwsCertficateManagerIntegrationTest succeeds.

### CODE REVIEW

File: core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/SigningStage.java [hunk 1]
```diff
@@ -132,6 +141,8 @@ private SdkHttpFullRequest signRequest(SdkHttpFullRequest request, RequestExecut
 
             SdkHttpFullRequest signedRequest = measuredSign.left();
 
+            // TODO: This case does not apply to SigningStage as event stream operations are not supported by SyncClients that
```
Comment (cenedhryn) [inline core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/SigningStage.java:144]: Why don't we remove it?

Comment (gosar) [reply core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/SigningStage.java:144]: I was leaving old code as-is for sometime, until more testing is done. I might remove it before release.

Comment (cenedhryn) [reply core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/SigningStage.java:144]: Are there any particular test cases that would be written later that we wouldn't remember to write if this were to be removed? I'd think that the tests would show if this piece of code was missing. It seems there are a lot of TODOs still to resolve before release... but if we're planning an activity to just go through all those and resolve them after general code complete, that should work too.

**Review by cenedhryn — APPROVED** [review]

File: core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/SigningStage.java (hunk 1 above)
Comment (gosar) [reply core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/SigningStage.java:144]: Yeah, going through all TODOs is in my delivery plan for after general code complete.

**Pull request merged**